src/main/java/org/synyx/urlaubsverwaltung/application/web/ApplicationForLeaveFormViewController.java

```diff
@@ -145,6 +151,76 @@ public String newApplication(@ModelAttribute("application") ApplicationForLeaveF
+
+        final Optional<Application> maybeApplication = applicationInteractionService.get(applicationId);
+        if (maybeApplication.isEmpty()) {
+            throw new UnknownApplicationForLeaveException(applicationId);
+        }
+
+        final Application application = maybeApplication.get();
+        if (application.getStatus().compareTo(WAITING) != 0) {
+            return "application/app_notwaiting";
+        }
+
+        final ApplicationForLeaveForm applicationForLeaveForm = mapToApplicationForm(application);
+        final Person person = personService.getSignedInUser();
+
+        final Optional<Account> holidaysAccount = accountService.getHolidaysAccount(Year.now(clock).getValue(), person);
+        if (holidaysAccount.isPresent()) {
+            prepareApplicationForLeaveForm(person, applicationForLeaveForm, model);
+        }
+        model.addAttribute("noHolidaysAccount", holidaysAccount.isEmpty());
+
+        model.addAttribute("application", applicationForLeaveForm);
+
+        return "application/app_form";
+    }
+
+    @PostMapping("/application/{applicationId}")
+    public String sendEditApplicationForm(@PathVariable("applicationId") Integer applicationId,
+                                          @ModelAttribute("application") ApplicationForLeaveForm appForm, Errors errors,
+                                          Model model, RedirectAttributes redirectAttributes) throws UnknownApplicationForLeaveException {
+
+        final Optional<Application> maybeApplication = applicationInteractionService.get(applicationId);
+        if (maybeApplication.isEmpty()) {
+            throw new UnknownApplicationForLeaveException(applicationId);
+        }
+
+        final Application application = maybeApplication.get();
+        if (application.getStatus().compareTo(WAITING) != 0) {
+            return "application/app_notwaiting";
+        }
+
+        appForm.setId(application.getId());
+        applicationForLeaveFormValidator.validate(appForm, errors);
+
+        if (errors.hasErrors()) {
+            prepareApplicationForLeaveForm(appForm.getPerson(), appForm, model);
+            if (errors.hasGlobalErrors()) {
+                model.addAttribute("errors", errors);
+            }
+            LOG.info("edit application ({}) has errors: {}", appForm, errors);
+            return "application/app_form";
+        }
+
+        final Application editedApplication = merge(application, appForm);
+        final Application savedApplicationForLeave;
+        final Person signedInUser = personService.getSignedInUser();
+        try {
+            savedApplicationForLeave = applicationInteractionService.edit(editedApplication, signedInUser, Optional.ofNullable(appForm.getComment()));
+        } catch (EditApplicationForLeaveNotAllowedException e) {
+            return "application/app_notwaiting";
```

Comment:
why `app_notwaiting` as template name?

Reply:
or we can replace this with in page feedback like this
![2020-10-30_14-52](https://user-images.githubusercontent.com/142107/97713092-aa1c7700-1abf-11eb-8e7e-f208b9d70e36.png)

Reply:
replaced